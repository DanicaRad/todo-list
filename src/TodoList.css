.todo-list {
  list-style-type: none;
  margin: 1em auto;
}