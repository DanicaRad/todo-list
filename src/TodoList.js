import { useState } from "react";
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';
import './TodoList.css';

const TodoList = () => {
  const [todos, setTodos] = useState([]);

  const addTodo = newTodo => {
    setTodos(todos => [...todos, newTodo])
  }

  const removeTodo = (id) => {
    setTodos(todos.filter(todo => todo.id !== id))
  }

  return (
    <div>
      <h3>New Todo List</h3>
      <NewTodoForm makeTodo={addTodo}/>
      <ul className="todo-list">
        {todos.map(({id, todo}) =>
          <Todo
            id={id}
            key={id}
            todo={todo}
            removeTodo={removeTodo}
          />
        )}
      </ul>
    </div>
  )
}

export default TodoList;