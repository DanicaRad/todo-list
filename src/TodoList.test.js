import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

function addTodo(todoList, todo='test todo item') {
  const todoInput = todoList.getByLabelText("Task:");
  fireEvent.change(todoInput, {target: {value: todo}});
  const button = todoList.getByText("Add Task");
  fireEvent.click(button);
};

it("renders without crashing", function() {
  render(<TodoList/>);
});

it("matches snapshot", function() {
  const {asFragment} = render(<TodoList/>);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo", function() {
  const todoList = render(<TodoList/>);

  expect(todoList.queryByText("X")).not.toBeInTheDocument();

  addTodo(todoList);

  const removeButton = todoList.getByText("X");
  expect(removeButton).toBeInTheDocument();
  expect(todoList.getByText("test todo item")).toBeInTheDocument();
  expect(todoList.getAllByDisplayValue("")).toHaveLength(1);
});

it("can remove a todo", function() {
  const todoList = render(<TodoList/>);
  addTodo(todoList);

  const removeButton = todoList.getByText("X");

  fireEvent.click(removeButton);
  expect(removeButton).not.toBeInTheDocument();
});