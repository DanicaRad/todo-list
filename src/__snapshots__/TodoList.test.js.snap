// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      New Todo List
    </h3>
    <div
      class="todo-form"
    >
      <form>
        <label
          for="todo"
        >
          Task:
        </label>
        <input
          id="todo"
          name="todo"
          type="text"
          value=""
        />
        <button>
          Add Task
        </button>
      </form>
    </div>
    <ul
      class="todo-list"
    />
  </div>
</DocumentFragment>
`;
